File: spinners/elixir/spinner.exs
#! /usr/bin/env elixir

defmodule Spinner do
  @spinner_chars String.graphemes("⠇⠋⠙⠸⠴⠦")
  def spin(msg, char_idx \\ 0) do
    char = Enum.at(@spinner_chars, char_idx)
    status = "\r#{char} #{msg}"

    receive do
      {super_pid, :computed} ->
        blanks = String.duplicate(" ", String.length(status))
        IO.write("\r#{blanks}\r")
        send(super_pid, :done)
    after
      100 ->
        IO.write(status)
        spin(msg, rem(char_idx + 1, length(@spinner_chars)))
    end
  end

  def slow_function() do
    Process.sleep(3000)
    42
  end

  def supervisor() do
    pid = spawn(fn -> spin("thinking!") end)
    result = slow_function()
    send(pid, {self(), :computed})

    receive do
      :done -> result
    end
  end

  def main() do
    result = supervisor()
    IO.puts("Answer: #{result}")
  end
end

Spinner.main()

File: spinners/go/spinner.go
package main

import (
	"fmt"
	"strings"
	"time"
)

func spin(msg string, computation <-chan bool, spinning chan<- bool) {
	spinner_chars := []rune(`⠇⠋⠙⠸⠴⠦`)
	i := 0
	status := ""
	repeat := true
	for repeat {
		select {
		case <- computation:
			repeat = false
		default:
			time.Sleep(100 * time.Millisecond)
			char := spinner_chars[i]
			i = (i + 1) % len(spinner_chars)
			status = fmt.Sprintf("\r%c %s", char, msg)
			fmt.Print(status)
		}
	}
	fmt.Printf("\r%s\r", strings.Repeat(" ", len(status)))
	spinning <- false
}

func slow_function() int {
    time.Sleep(3 * time.Second)
	return 42
}

func supervisor() int {
	computation := make(chan bool)
	spinning := make(chan bool)
	go spin("thinking!", computation, spinning)
	result := slow_function()
	computation <- true
	<- spinning
	return result
}

func main() {
	result := supervisor()
	fmt.Println("Answer:", result)
}
